lib/utils.js

```diff
@@ -99,52 +102,35 @@ var getStringifiedStyles = function(rule) {
  */
 
 exports.getPreservedText = function(css, options) {
-  var rules = cssom.parse(css).cssRules || [];
-  var preserved = [];
-
-  for (var i = 0, l = rules.length; i < l; i++) {
-    /* CSS types
-      STYLE: 1,
-      IMPORT: 3,
-      MEDIA: 4,
-      FONT_FACE: 5,
-    */
-
-    if (options.fontFaces && rules[i].type === cssom.CSSFontFaceRule.prototype.type) {
-      var fontFace = [ '' ];
-      fontFace.push('@font-face {');
-      fontFace = fontFace.concat(getStringifiedStyles(rules[i]));
-      fontFace.push('}');
-
-      if (fontFace.length) {
-        preserved.push(fontFace.length ? fontFace.join(os.EOL) + os.EOL : '');
-      }
-    }
-
-    if (options.mediaQueries && rules[i].type === cssom.CSSMediaRule.prototype.type) {
-      var query = rules[i];
-      var queryString = [];
-
-      queryString.push(os.EOL + '@media ' + query.media[0] + ' {');
-
-      for (var ii = 0, ll = query.cssRules.length; ii < ll; ii++) {
-        var rule = query.cssRules[ii];
-
-        if (rule.type === cssom.CSSStyleRule.prototype.type
-            || rule.type === cssom.CSSFontFaceRule.prototype.type) {
-          queryString.push('  '
-            + (rule.type === cssom.CSSStyleRule.prototype.type ? rule.selectorText : '@font-face') + ' {');
-          queryString = queryString.concat(getStringifiedStyles(rule));
-          queryString.push('  }');
-        }
-      }
-
-      queryString.push('}');
-      preserved.push(queryString.length ? queryString.join(os.EOL) + os.EOL : '');
+  var parsed = mensch.parse(css, {position: true, comments: true});
+  var rules = typeof parsed.stylesheet != 'undefined' && parsed.stylesheet.rules ? parsed.stylesheet.rules : [];
+  var preserved = css;
+  var preserved2 = [];
+  var lastStart = null;
+
+  for (var i = rules.length - 1; i >= 0; i--) {
+    if (options.fontFaces && rules[i].type === 'font-face') {
+      // preserve
+      preserved2.push(mensch.stringify({ stylesheet: { rules: [ rules[i] ] }}, { comments: false, indentation: '    ' }).replace(/}/,os.EOL+'}'));
+    } else if (options.mediaQueries && rules[i].type === 'media') {
+      // preserve
+      preserved2.push(mensch.stringify({ stylesheet: { rules: [ rules[i] ] }}, { comments: false, indentation: '  ' }));
+    } else {
+      // remove
+      preserved = removeStyle(preserved, rules[i].position.start, lastStart, 0, 0, 0, '');
+    // TODO +os.EOL?
     }
+    lastStart = rules[i].position.start;
+  }
+  // preserved works by removing unwanted stuff, preserved2 by generating a new style with "stuff to keep"
+  // We have to decide what is our strategy (the preserved2 produces the same output of the cssom based version <= 2.0.0)
+  if (false) {
```

Comment:
Yes, see the comment above (I just don't know what kind of behaviour you prefer, so I implemented both of them and kept an if to enable the alternative way.

Again I don't consider this code "ready to be released" but depending on the way you are used to collaborate this could be merged while we keep working on other aspects. I just wanted to share what I'm doing ASAP, so we stay more aligned.